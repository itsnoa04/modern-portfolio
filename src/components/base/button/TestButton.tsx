export interface TestButtonProps {
  children: React.ReactNode;
}

const TestButton: React.FC<TestButtonProps> = ({ children }) => {
  return <button type="button">{children}</button>;
};

export default TestButton;
