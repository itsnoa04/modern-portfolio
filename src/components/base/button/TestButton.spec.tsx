import React from "react";

import { render, screen } from "@testing-library/react";

import TestButton from "./TestButton";

describe("TestButton", () => {
  it("renders children", () => {
    render(<TestButton>hi</TestButton>);
    expect(screen.getByText("hi")).toBeVisible();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<TestButton>hi</TestButton>);
    expect(asFragment()).toMatchSnapshot();
  });
});
