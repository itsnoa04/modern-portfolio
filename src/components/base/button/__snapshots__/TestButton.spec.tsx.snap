// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    hi
  </button>
</DocumentFragment>
`;
